### PR TITLE
Implement Natural Death Function

### PR DESCRIPTION
**Pull request by FMorgan-111**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -38,7 +38,27 @@ def reproduce(population: List[Deer], params: ModelParameters):
...
+        elif 5 <= deer.age < 12:
+            mortality_rate = params.matureMortalityRate
+        else:
+            mortality_rate = params.oldMortalityRate
+        
+        # Determine if the deer survives based on mortality rate
+        if random() > mortality_rate:
+            survivors.append(deer)
+    
+    return survivors
```
Comment (birchtree02) [inline src/main.py:77]: Screenshot 2024-11-07 at 12 30 36
This function should be based on the death algorithm from the paper.

File: src/main.py [hunk 2]
```diff
@@ -68,13 +88,21 @@ def __init__(
...
         self.sampleSpace = sampleSpace,                 # S
         self.maxCapacityImpact = maxCapacityImpact,     # c
         self.capacityCurveSlope = capacityCurveSlope,   # a
         self.huntingLimit = huntingLimit,               # l
         self.initialIndividuals = initialIndividuals,   # i_init
         self.maximumIndividuals = maximumIndividuals    # i_max
+        self.calfMortalityRate = calfMortalityRate
+        self.youngMortalityRate = youngMortalityRate
+        self.matureMortalityRate = matureMortalityRate
+        self.oldMortalityRate = oldMortalityRate
```
Comment (birchtree02) [inline src/main.py]: Screenshot 2024-11-07 at 12 32 32
Mortality rates are not defined by parameters, but by this algorithm from the paper.

**Comment by birchtree02** [comment]
Screenshot 2024-11-07 at 12 32 32

Mortality rates are not defined by parameters, but by this algorithm from the paper.

File: src/main.py [hunk 3]
```diff
@@ -36,15 +36,43 @@ def reproduce(population: List[Deer], params: ModelParameters):
 
     return population + newDeer
 
+def adjustMortalityRate(base_mortality: float, max_cap_impact: float, cap_curve_slope: float, inow: int, imax: int) -> float:
```
Comment (birchtree02) [inline src/main.py:50]: this looks good

File: src/main.py [hunk 4]
```diff
@@ -36,15 +36,43 @@ def reproduce(population: List[Deer], params: ModelParameters):
...
-    pass
+def naturalDeath(population: List[Deer], params: ModelParameters):
+    survivors = []
+    inow = len(population)  # Current population size
+    imax = params.maximumIndividuals  # Maximum carrying capacity
+
+    for deer in population:
+        # Determine base mortality rate based on age
+        if deer.age == 0:
+            base_mortality = 0.2  # Base mortality rate for age 0
```
Comment (birchtree02) [inline src/main.py]: Where did you get these values for base mortality?

Comment (FMorgan-111) [reply src/main.py]: it is initial value i assign here, since it will go through the function adjustmorality

**Review by birchtree02 — APPROVED** [review]
Ready to merge

**Pull request merged**